core/Plugin/Visualization.php

```diff
@@ -804,6 +817,67 @@ public function buildApiRequestArray()
             unset($request['disable_queued_filters']);
         }
 
+        if ($this->isComparing()) {
+            $request['compare'] = '1';
+        }
+
         return $request;
     }
+
+    /**
+     * TODO
+     * @return bool
+     */
+    public function isComparing()
+    {
+        if (!$this->supportsComparison()) {
+            return false;
+        }
+
+        $request = $this->getRequestArray();
+
+        return !empty($request['compareSegments'])
+            || !empty($request['comparePeriods'])
+            || !empty($request['compareDates']);
+    }
+
+    /**
+     * TODO
+     * @return bool
+     */
+    public function supportsComparison()
+    {
+        return false;
+    }
+
+    private function getRequestArray()
+    {
+        if (empty($this->cachedRequestArray)) {
```

Comment:
Not 100% sure about this cache. Is it needed? Should the RequestConfig change, then the requestArray would change as well or not and the cache would need to be invalidated?